### PR TITLE
Fix command line support in linux tv-casting-app

### LINKED ISSUE #24214: [BUG] tv-casting-app, command line arguments are broken
### Reproduction steps

tv-casting-app works similar to chip-tool in that cluster commands can be passed as command line arguments and these will be invoked by tv-casting-app

Recent changes to auto-load fabrics has broken this feature

### Bug prevalence

always

### GitHub hash of the SDK that was being used

master

### Platform

core

### Platform Version(s)

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by chrisdecenzo**
Fixes #24214

tv-casting-app works similar to chip-tool in that cluster commands can be passed as command line arguments and these will be invoked by tv-casting-app

Recent changes to auto-load fabrics has broken this feature

### CODE REVIEW

File: examples/tv-casting-app/tv-casting-common/src/CastingServer.cpp [hunk 1]
```diff
@@ -168,6 +168,7 @@ CastingServer::GetDiscoveredCommissioner(int index, chip::Optional<TargetVideoPl
 void CastingServer::ReadServerClustersForNode(NodeId nodeId)
 {
     ChipLogProgress(NotSpecified, "ReadServerClustersForNode nodeId=0x" ChipLogFormatX64, ChipLogValueX64(nodeId));
+    Init();
```
Comment (andy31415) [inline examples/tv-casting-app/tv-casting-common/src/CastingServer.cpp:171]: Init seems to be `CHIP_ERROR CastingServer::Init`. Could we add error checking here and in the other newly added init call sites? Even if it is just a `log on error`

File: examples/tv-casting-app/tv-casting-common/src/TargetVideoPlayerInfo.cpp [hunk 2]
```diff
@@ -40,7 +40,10 @@ CHIP_ERROR TargetVideoPlayerInfo::Initialize(NodeId nodeId, FabricIndex fabricIn
         mIpAddress[i] = ipAddress[i];
     }
 
-    chip::Platform::CopyString(mDeviceName, chip::Dnssd::kMaxDeviceNameLen + 1, deviceName);
+    if (deviceName)
+    {
+        chip::Platform::CopyString(mDeviceName, chip::Dnssd::kMaxDeviceNameLen + 1, deviceName);
```
Comment (andy31415) [inline examples/tv-casting-app/tv-casting-common/src/TargetVideoPlayerInfo.cpp:45]: mDeviceName is already `TargetVideoPlayerInfo.h:    char mDeviceName[chip::Dnssd::kMaxDeviceNameLen + 1]` so CopyString should be a be to auto-deduce size. Could we use `chip::Platform::CopyString(mDeviceName, deviceName)` ?

**Review by sharadb-amazon — APPROVED** [review]

File: examples/tv-casting-app/tv-casting-common/src/TargetVideoPlayerInfo.cpp (hunk 2 above)
Comment (chrisdecenzo) [reply examples/tv-casting-app/tv-casting-common/src/TargetVideoPlayerInfo.cpp]: addressed

File: examples/tv-casting-app/tv-casting-common/src/CastingServer.cpp (hunk 1 above)
Comment (chrisdecenzo) [reply examples/tv-casting-app/tv-casting-common/src/CastingServer.cpp]: added logging of any error encountered

**Review by andy31415 — APPROVED** [review]

**Pull request merged**